Fixing modelservice canary and destinationrule

rnjsxodyd90 opened the pull request:
The A4 assignment required two versions of both services with consistent routing (old→old, new→new), but model-service only had one version missing canary support.

rnjsxodyd90 committed:
fixing modelservice canary and destinationrule

rnjsxodyd90 committed:
Add Istio canary deployment testing instructions to README

rnjsxodyd90 committed:
fixed the readme, clearer for verification

rnjsxodyd90 committed:
Add Istio sidecar injection instructions to README

martin-gou commented:
I tested and below are some issues that I met:
1. 
```bash
(base)  ~/TUD_Q2/CS4295-DevOps/assignment/operation/ [fixing-destinationrule-and-canarymodelservice] KUBECONFIG=./kubeconfig kubectl exec deploy/app-stable -c istio-proxy -- \
  pilot-agent request GET config_dump | grep -A5 '"cluster": "outbound|8081|' | head -10
Error from server (BadRequest): container istio-proxy is not valid for pod app-stable-6577d4698c-zm7b4
```
and also, when I run in local helm. This command return me failture
<img width="1280" height="720" alt="image" src="https://github.com/user-attachments/assets/fd1c9ec3-6a5d-4c3d-b722-b7cf3b4ec3be" />


2. Getting into http://sms.local/sms/, and send a few request, and I do 
```bash
KUBECONFIG=./kubeconfig kubectl logs deploy/model-service-stable -c model-service --tail=5 | grep POST
KUBECONFIG=./kubeconfig kubectl logs deploy/model-service-canary -c model-service --tail=5 | grep POST
```
But I found both model-service have incoming visit. Seems like `old→old, new→new` is not true.

rnjsxodyd90 commented:
Ill check this.

rnjsxodyd90 committed:
Fix model service canary routing for consistent version routing

Adding version specific services, updating app deployments to use correct MODEL_HOST, updated model virtualservice routing and updated the README

rnjsxodyd90 committed:
Update README with A3 Prometheus dashboard and A4 Istio documentation

rnjsxodyd90 committed:
Merge branch 'main' into fixing-destinationrule-and-canarymodelservice

martin-gou reviewed (approved):
Good to go
